src/util.ts

```diff
@@ -71,10 +71,11 @@ export const getTracesApiKey = (options?: HoneycombOptions) => {
 
 export const getSampleRate = (options?: HoneycombOptions) => {
   if (
-    // sample rate must be a whole integer greater than 0
-    options?.sampleRate &&
-    options?.sampleRate > 0 &&
-    Number.isSafeInteger(options?.sampleRate)
+    // must be a whole positive integer
+    options?.sampleRate !== null &&
+    options?.sampleRate !== undefined &&
```

Comment:
Falsy values in JS are wild. This is necessary as `options?.sampleRate` will be falsy and return the default sample rate (see [typescript playground here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAIgUQGIEECqAZAKgfQMooCyAChgjgEopYIwC8MAjANwBQrAlmFAKYBOAMwCGwHjAAS4HgE9QAWwBGAeQAOUDuAgwA3qxj6YEIXJUAbHhSG8A-AC4YYAK6L+bAL7tQkWAHMeUPGMzCysxBgAKEDUNSDsJKVkQRVV1TQBKegA+HT0DDgEYcNyDAwB6Upg5R2gYBTEhGAB3AAsQcxgVEAgOdQA3MS5ePz5ikpgo1NiAOiMTc0teGABCOgYnU1MYADIt0ZKJmIhrGaD50OXVmEcwABMeAS4eG+3dsYMDzWPZ4IWxTIYAAwvPYGAByzjqfCmHAggQEPAAktweMNItFPic5iFeBlVgwoHxHDxRhldG99Hx-I4+GBxujpt8zrw2CUPCVKVBqbTEKhMLgCCQyJRqAh3GxWF4an4AqdsTwkEJTBBpPRCh9YvZJGAZPJlPSILjsmS8gUiuSYOVKtVYHUYA0Wm0xJ1un0BsjhiD9Oqjpifucdp66ZMfYy5TB-jAgQHzeCXFCYXDEe7+Gjg19Zb9cZcCUSSTlzRyuUHDumsb8WQY2QZCzT4Mh0Nh8ERSOQqDQxZ5NLBvQgTFBpJqErqUodVdo3GxJd39QDB9rEsl9WPDBnQvYgRPO5BHVNTCAfOEAAbSwJl0Kpw69tTSNL2AAk2hPq94F80V-7aTch4ylsYEs0O57gex7+Kefovt6AK3jAD5PmeEEzp+34WhUAL-tu5i7vuR5weB8qKsqr6QO+N73o+oHPvhSrSEREAkUhP4VH+kqAdhIEyvBVGEZB0GwRRnEKtRtFQV+jFMEAA))

Reply:
`Number.isSafeInteger` should return false for null & undefined as well as non-integer numbers, so i think you could shorten this to:

```tsx
if (Number.isSafeInteger(options?.sampleRate) && options.sampleRate >= 0) {
 return options.sampleRate
}
```

Reply:
I guess it may be a TypeScript thing, but I can't get that second part of the equation in there, because I get an error of `'options.sampleRate' is possibly 'undefined'.`

Reply:
(if the `Number.isSafeInteger(options?.sampleRate)` check passes, typescript knows options.sampleRate both exists & is a number, so you don't need the conditional chaining -- `options?.sampleRate` -- for any thing that follows)

Reply:
This is what you mean right? I could definitely be missing something still, but I think TypeScript is not happy with this.

![sampleRate could be undefined](https://github.com/honeycombio/honeycomb-opentelemetry-web/assets/29520003/7638f5a7-fe3a-4ec5-8495-c1f99d97683a)

Reply:
Okay! Yeah it seems like that's how it should work for exactly why you mention it. But it must be something about the static analysis of the code or something.

Reply:
ah... i'm used to writing in a codebase w/out `strictNullChecks` enabled: 

[sloppy ts setting](https://www.typescriptlang.org/play?strict=false&strictNullChecks=false&alwaysStrict=false#code/MYewdgzgLgBAIgUQGIEECqAZAKgfQMooCyAChgjgEopYIwC8MAjANwBQrAlmFAKYBOAMwCGwHjAAS4HgE9QAWwBGAeQAOUDuAgwA3qxj6YEIXJUAbHhSG8A-AC4YYAK6L+bAL7tQkWAHMeUPGMzCysxBgAKEDUNSDsJKVkQRVV1TQBKegA+HT0DDgEYcNyDAwB6Upg5R2gYBTEhGAB3AAsQcxgVEAgOdQA3MS5ePz5ikpgo1NiAOiMTc0teGABCOgYnU1MYADIt0ZKJmIhrGaD50OXVmEcwABMeAS4eG+3dsYMDzWPZ4IWxTIYAAwvPYGAByzjqfCmHAggQEPAAktweMNItFPic5iFeBlVgwoHxHDxRhldG99Hx-I4+GBxujpt8zrw2CUPCVKVBqbTEKhMLgCCQyJRqAh3GxWF4an4AqdsTwkEJTBBpPRCh9YvZJGAZPJlPSILjsmS8gUiuTwS4oTC4Yjkaj1UdMT9QhkdnTJhAnUy-oCSTlyTAOVz3YcvXKWQY2QYgzT4Mh0Nh8ERSOQqDQxZ5NLAHQgTFBpJqErqUodVdo3GxJdn9QDC9rEsl9WXDLLfvYgRXM5A2jwpqYQD5wgADaWBLG-NEe3NqaRpewAEm0o9boUnh2n+bSbiHGXKTAlmh7fYHw+X49XDoBc5gi7Pzt4a80V+3u4qAIP3fMx8HI-8Y-v8qKsqj6QBus4Lkuf4rrwCpKtIIEQGBW47jAe6MB+EBHv2P53t6sHAZe163lB54wUB8GES+qEVIwQA)

Reply:
sorry for the red herring!

Reply:
😄  "sloppy ts setting". No worries, if we decide we want that setting set to false later we can totally do that and simplify this code. I don't have super strong feels, though tend to lean toward stricter type checking to save me from myself (also lazy, this one is a default). But this is one of those scenarios that definitely gets people to shake their fist at TypeScript so I totally get it ha.

Reply:
i don't know if this is any better, but adding a proper typecheck got strict typescript's begrudging acceptance:

```ts
const getSampleRateFalsy = (options?: HoneycombOptions) => {
    if (typeof options?.sampleRate !== "number" || !Number.isSafeInteger(options.sampleRate) || options.sampleRate < 0) {
        return DEFAULT_SAMPLE_RATE;
    }

    return options.sampleRate;
};
```

Reply:
I'm leaning toward this actually, also because I like that early return style too (you've mentioned in another PR, and I had something like this for a minute before switching back).

Reply:
I changed my mind again on the early return only because I wasn't a fan of all the negations... BUT still took your suggestion to use the typeof check! Thanks for going on this journey with me! https://github.com/honeycombio/honeycomb-opentelemetry-web/pull/72/commits/d1e235e0dd9d35db66c24012b65ee1d351222649